scripts/lint/ourlint.js

```diff
@@ -333,6 +341,106 @@ ${invalids.map((icon) => `${format(icon)} ${findPositon(expectedOrder, icon)}`).
 
 		return errors.join('\n') || undefined;
 	},
+
+	/* Ensure that titles constraints are enforced. */
+	checkTitles(icons) {
+		const titles = new Set();
+		const duplicateTitles = [];
+		for (const icon of icons) {
+			const {title, slug} = icon;
+
+			// Titles of icons that do not have slug must be unique.
+			if (slug === undefined) {
+				if (titles.has(title)) {
+					duplicateTitles.push(title);
+				} else {
+					titles.add(title);
+				}
+			}
+		}
+
+		const errors = [];
+		if (duplicateTitles.length > 0) {
+			const message =
+				`Found duplicate title for icons that do not have slug: ${duplicateTitles.join(', ')}.` +
+				`\nPlease, ensure that all titles are unique or these icons have proper slugs.`;
+			errors.push(message);
+		}
+
+		return errors.join('\n') || undefined;
+	},
+
+	/* Ensure that slugs constraints are enforced. */
+	async checkSlugs(icons) {
+		const errors = [];
+
+		for (const icon of icons) {
+			const {slug, title} = icon;
+
+			if (slug === undefined) {
+				continue;
+			}
+
+			// Custom slugs must be necessary. If a title is converted to a slug
+			// and it is the same as the slug, then it is not necessary.
+			if (titleToSlug(title) === slug) {
+				errors.push(
+					`Icon "${title}" has a slug "${slug}" that is the same as the slug inferred from its title.` +
+						' Please, remove the slug or change the title.',
```

Comment:
We probably shouldn't be inviting people to change the `title` of an icon as it could be implied that altering a brand's name is permissible.